GitHub Action errors building the master branch

<!--
     Hello! 👋
     Thank you for opening an issue on PouchDB.
     There is also a 3-Minute read, on how to format your issue:
     https://guides.github.com/features/mastering-markdown/
     This template is optional, but it may help us fix your issue faster.
-->
### Issue
GitHub Actions build of the master branch are showing error Annotations but ultimately the build passes with a green check. It appears as if the first try fails on a test failure, then when the retry runs it passes, so the whole build is marked as passing. However when the retry happens nothing is logged which makes me think it's not actually running anything at all.

Logs:
```
// lots of errors

  [29]</</</testUtils.cleanup@utils-bundle.js:15383:7
  @test.issue3646.js:15:17
  



npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! pouchdb-monorepo@7.0.0-prerelease test-browser: `npm run build-test && node ./bin/test-browser.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the pouchdb-monorepo@7.0.0-prerelease test-browser script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2023-07-24T22_34_18_786Z-debug.log
npm ERR! Test failed.  See above for more details.
Error: Process completed with exit code 1.

// First retry

Run npm test
  npm test
  shell: /usr/bin/bash -e {0}
  env:
    NODE_VERSION: 14
    TEST_HOST: localhost
    CLIENT: firefox
    SERVER: pouchdb-server

> pouchdb-monorepo@7.0.0-prerelease test /home/runner/work/pouchdb/pouchdb
> ./bin/run-test.sh

// Cleanup
```

Between the ./bin/run-test.sh and the // Cleanup it should actually run the tests...

Screenshot:

https://user-images.githubusercontent.com/191496/255793747-357c2268-2f88-4e66-b460-d22ae33ab45d.png

Example build: https://github.com/pouchdb/pouchdb/actions/runs/5650289637

There are other examples going back 7 months (or more) so this isn't a recent change.


### Info
- Environment: N/A
- Platform: GitHub
- Adapter: All
- Server: N/A

### Reproduce
Seems to happen frequently on `master` though I haven't seen it on PR builds.

--- a/bin/run-test.sh
+++ b/bin/run-test.sh
@@ -20,8 +20,7 @@ pouchdb-setup-server() {
   # in CI, link pouchdb-servers dependencies on pouchdb
   # modules to the current implementations
   if [ -d "pouchdb-server-install" ]; then
-    # pouchdb server already running
-    exit 0
+    rm -rf pouchdb-server-install
   fi
   mkdir pouchdb-server-install
   cd pouchdb-server-install

